From 4bceb43366afb800be1b1ed72cc1f111e707b3ac Mon Sep 17 00:00:00 2001
From: Daniel Moore <dtmoore@motorola.com>
Date: Tue, 18 Nov 2014 14:06:29 -0600
Subject: [PATCH 095/959] IKLOCSEN-1586:drivers:misc:stml0xx: Disable SH
 logging for user

Only enable SH debug logging for non-user builds

Change-Id: I888e97e1c6a70ce7a3477609ecfd19ef6af3098c
Signed-off-by: Dan Moore <dtmoore@motorola.com>
Reviewed-on: http://gerrit.mot.com/691322
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Robert Wiegand <Robert.P.Wiegand@motorola.com>
Reviewed-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/misc/stml0xx/Kconfig         |  7 +++++++
 drivers/misc/stml0xx/stml0xx_reset.c | 11 +++++++++++
 include/linux/stml0xx.h              |  9 +++++++++
 3 files changed, 27 insertions(+)

diff --git a/drivers/misc/stml0xx/Kconfig b/drivers/misc/stml0xx/Kconfig
index caaff764f94..5ec96202238 100644
--- a/drivers/misc/stml0xx/Kconfig
+++ b/drivers/misc/stml0xx/Kconfig
@@ -1,3 +1,10 @@
+config SENSORHUB_DEBUG_LOGGING
+	bool "sensorhub debug logging"
+	default n
+	help
+	  Say yes here if you wish to enable debug
+	  logging from the sensorhub.
+
 config SENSORS_STML0XX
 	tristate "STML0XX Sensor Processor"
 	default n
diff --git a/drivers/misc/stml0xx/stml0xx_reset.c b/drivers/misc/stml0xx/stml0xx_reset.c
index 52338464495..de3f0d407b7 100644
--- a/drivers/misc/stml0xx/stml0xx_reset.c
+++ b/drivers/misc/stml0xx/stml0xx_reset.c
@@ -203,6 +203,17 @@ void stml0xx_initialize_work_func(struct work_struct *work)
 		}
 	}
 
+#ifdef CONFIG_SENSORHUB_DEBUG_LOGGING
+	buf[0] = SH_LOG_DEBUG;
+	err = stml0xx_spi_send_write_reg_reset(SH_LOG_LEVEL_REG, buf,
+		1, RESET_NOT_ALLOWED);
+	if (err < 0) {
+		dev_err(&ps_stml0xx->spi->dev,
+			"Unable to write sh log level");
+		ret_err = err;
+	}
+#endif
+
 	err = stml0xx_led_set_reset(&ps_stml0xx->led_cdev,
 			RESET_NOT_ALLOWED);
 	if (err < 0)
diff --git a/include/linux/stml0xx.h b/include/linux/stml0xx.h
index df63c6c2547..56fc78e02d8 100644
--- a/include/linux/stml0xx.h
+++ b/include/linux/stml0xx.h
@@ -319,6 +319,13 @@ enum lowpower_mode {
 	LOWPOWER_ENABLED
 };
 
+enum sh_log_level {
+	SH_LOG_DISABLE,
+	SH_LOG_ERROR,
+	SH_LOG_VERBOSE,
+	SH_LOG_DEBUG
+};
+
 struct stm_response {
 	/* 0x0080 */
 	unsigned short header;
@@ -410,6 +417,8 @@ struct stm_response {
 #define NFC                             0x4D
 #define SIM                             0x4E
 
+#define SH_LOG_LEVEL_REG                0x55
+
 #define ALGO_CFG_ACCUM_MODALITY         0x5D
 #define ALGO_REQ_ACCUM_MODALITY         0x60
 #define ALGO_EVT_ACCUM_MODALITY         0x63
-- 
2.11.0

